index.js

```diff
@@ -3209,27 +3209,25 @@
   }
   S.find = def('find', {}, [Pred(a), $.Array(a), $Maybe(a)], find);
 
-  //# pluck :: Accessible a => String -> Array a -> Array b
+  //# pluck :: (Accessible a, Functor f) => String -> f a -> f b
   //.
   //. Combines [`map`](#map) and [`prop`](#prop). `pluck(k, xs)` is equivalent
   //. to `map(prop(k), xs)`.
   //.
   //. ```javascript
   //. > S.pluck('x', [{x: 1}, {x: 2}, {x: 3}])
   //. [1, 2, 3]
+  //.
+  //. > S.pluck('x', S.Just({x: 1, y: 2, z: 3}))
+  //. Just(1)
   //. ```
   function pluck(key, xs) {
-    return xs.map(function(x, idx) {
-      if (key in Object(x)) return x[key];
-      throw new TypeError('‘pluck’ expected object at index ' + idx +
-                          ' to have a property named ‘' + key + '’; ' +
-                          Z.toString(x) + ' does not');
-    });
+    return Z.map(function(obj) { return prop(key, obj); }, xs);
```

Comment:
`S.prop` is curried, but here [`prop`][1] refers to the internal implementation which is not curried.


[1]: https://github.com/sanctuary-js/sanctuary/blob/v0.12.2/index.js#L3129-L3133